openlibrary/plugins/upstream/utils.py

```diff
@@ -136,8 +136,7 @@ def render_component(name, attrs=None, json_encode=True):
     for (key, val) in attrs.items():
         if json_encode and isinstance(val, dict) or isinstance(val, list):
             val = json.dumps(val)
-        attrs_str += ' %s="%s"' % (key, val.replace('"', "'"))
-
+        attrs_str += f' {key}=\'{val}\''
```

Comment:
Actually, if we want to keep this simple, instead of trying to manually escape the json, let's just URL encode the whole string. I believe `urllib.parse.quote` should do the trick? And then in JS land call `decodeURIComponent` before `JSON.parse` :+1:

```suggestion
        attrs_str += f' {key}="{urllib.parse.quote(val)}"'
```

Reply:
Oh, we actually only want to do that in the if statement after `json.dumps`!

Reply:
added and it's working nicely